### PR TITLE
Harbor upgrade and data migration script

### PR DESCRIPTION
**Pull request by andrewtchin**
Towards https://github.com/vmware/vic/issues/4726

### CODE REVIEW

File: installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -euf -o pipefail
+
+data_mount="/data/harbor"
+cfg="${data_mount}/harbor.cfg"
+harbor_backup="/data/harbor_backup"
+harbor_migration="/data/harbor_migration"
+psc_token_file="/etc/vmware/psc/harbor/token.properties"
```
Comment (anchal-agrawal) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: `tokens.properties`

Comment (andrewtchin) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: fixed

File: installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+  # Start docker service
+  systemctl start docker.service
+  sleep 2
+
+	local migrator_image="vmware/harbor-db-migrator:1.2-test"
+	local harbor_database="/data/harbor/database"
+
+  docker run -it --rm -e DB_USR=${DB_USER} -e DB_PWD=${DB_PASSWORD} -v ${harbor_database}:/var/lib/mysql -v ${harbor_backup}:/harbor-migration/backup ${migrator_image} backup
+	docker run -it --rm -e DB_USR=${DB_USER} -e DB_PWD=${DB_PASSWORD} -v ${harbor_database}:/var/lib/mysql ${migrator_image} up head
```
Comment (wy65701436) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: please add -e SKIP_CONFIRM=y to bypass the interactive with the upgrade shell script.

File: installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  # Overwrites ${harbor_migration}/harbor_projects.json if present
+  docker run -ti --rm -e DB_USR=${DB_USER} -e DB_PWD=${DB_PASSWORD} –e EXPORTPATH=/harbor_migration –v ${harbor_migration}:/harbor_migration -v ${harbor_database}:/var/lib/mysql ${migrator_image} export
```
Comment (wy65701436) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: the migrator image was packaged into the offline installer. You can call docker load to get the image from local tgz file. 

See this PR. https://github.com/vmware/harbor/pull/2926

File: installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  mkdir ${harbor_migration}
+
+  # Start docker service
+  systemctl start docker.service
+  sleep 2
+
+	local migrator_image="vmware/harbor-db-migrator:1.2-test"
+	local harbor_database="/data/harbor/database"
+
+  docker run -it --rm -e DB_USR=${DB_USER} -e DB_PWD=${DB_PASSWORD} -v ${harbor_database}:/var/lib/mysql -v ${harbor_backup}:/harbor-migration/backup ${migrator_image} backup
```
Comment (andrewtchin) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: That is done starting on line 203

File: installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh (hunk 2 above)
Comment (andrewtchin) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: fixed

File: installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh (hunk 3 above)
Comment (andrewtchin) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: That's what we do here right? https://github.com/andrewtchin/vic-product/blob/master/installer/packer/scripts/provision_harbor.sh#L48

File: installer/packer/scripts/harbor/start_harbor.sh [hunk 5]
```diff
@@ -35,13 +35,13 @@ function gc {
...
+    "$registry_image" garbage-collect /etc/registry/config.yml
 
   echo "===================================================="
 }
 
 gc_enabled=$(ovfenv --key registry.gc_enabled)
-if [ ${gc_enabled,,} == "true" ]; then
+if [ "${gc_enabled,,}" == "true" ]; then
   gc 2>&1 >> /var/log/harbor/gc.log || true
 fi
```
Comment (wy65701436) [inline installer/packer/scripts/harbor/start_harbor.sh]: I have an comment on line 48 about to launch Harbor, for the latest release it should be,
/usr/local/bin/docker-compose -f /etc/vmware/harbor/docker-compose.yml -f /etc/vmware/harbor/docker-compose.notary.yml -f /etc/vmware/harbor/docker-compose.clair.yml up

Comment (andrewtchin) [inline installer/packer/scripts/harbor/start_harbor.sh]: fixed

File: installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh [hunk 6]
```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/bash
```
Comment (andrewtchin) [inline installer/packer/scripts/harbor/upgrade_1.1_to_1.2.sh]: Output goes to stdout

**Pull request merged**